### PR TITLE
create a job for db trigger

### PR DESCRIPTION
**Pull request by sywhb**

### CODE REVIEW

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/queue-processor.ts [hunk 1]
```diff
@@ -34,6 +47,43 @@ export const getBackendQueue = async (): Promise<Queue | undefined> => {
   return backendQueue
 }
 
+export const createWorker = (connection: ConnectionOptions) =>
```
Comment (jacksonh) [inline packages/api/src/queue-processor.ts:50]: 🚀

File: packages/api/src/services/labels.ts [hunk 2]
```diff
@@ -218,12 +233,32 @@ export const deleteLabels = async (
...
+
+  if (!deleteResult.affected) {
+    return false
+  }
+
+  const data = libraryItemIds.map((libraryItemId) => ({
+    libraryItemId,
+    userId,
+  }))
+  await bulkEnqueueUpdateLabels(data)
```
Comment (jacksonh) [inline packages/api/src/services/labels.ts:251]: nice way of doing it

**Review by jacksonh — COMMENTED** [review]
So does this mean highlights and labels are not cached on the library item anymore?

**Comment by sywhb** [comment]
> So does this mean highlights and labels are not cached on the library item anymore?

no, they will still be cached if i understand the question correctly

**Pull request merged**